otherlibs/systhreads/st_stubs.c

```diff
@@ -371,10 +294,10 @@ static value caml_thread_new_descriptor(value clos)
     mu = caml_threadstatus_new();
     /* Create a descriptor for the new thread */
     descr = caml_alloc_small(3, 0);
-    Ident(descr) = Val_long(thread_next_ident);
+    Ident(descr) = Val_long(atomic_load_acq(&thread_next_id));
     Start_closure(descr) = clos;
     Terminated(descr) = mu;
-    thread_next_ident++;
+    atomic_fetch_add(&thread_next_id, +1);
```

Comment:
This is not atomic enough!  Two threads can still be created with the same ID (consider the interleaving: load / load / fetch_add / fetch_add).  Just do
```
    Ident(descr) = Val_long(atomic_fetch_add(&thread_next_id, +1));
```